include/openssl/core_names.h

```diff
@@ -94,6 +94,32 @@ extern "C" {
 #define OSSL_MAC_NAME_KMAC128       "KMAC128"
 #define OSSL_MAC_NAME_KMAC256       "KMAC256"
 
+/* KDF / PRF parameters */
+#define OSSL_KDF_PARAM_SECRET       "secret"    /* octet string */
+#define OSSL_KDF_PARAM_KEY          "key"       /* octet string */
+#define OSSL_KDF_PARAM_SALT         "salt"      /* octet string */
+#define OSSL_KDF_PARAM_PASSWORD     "password"  /* octet string */
+#define OSSL_KDF_PARAM_DIGEST       "digest"    /* utf8 string */
+#define OSSL_KDF_PARAM_MAC          "mac"       /* utf8 string */
+#define OSSL_KDF_PARAM_MAC_SIZE     "mac_size"  /* size_t */
+#define OSSL_KDF_PARAM_PROPERTIES   "properties" /* utf8 string */
+#define OSSL_KDF_PARAM_ITER         "iter"      /* unsigned int */
+#define OSSL_KDF_PARAM_MODE         "mode"      /* utf8 string */
+#define OSSL_KDF_PARAM_MODE_INT     "mode_int"  /* int */
+#define OSSL_KDF_PARAM_PKCS5        "pkcs5"     /* int */
+#define OSSL_KDF_PARAM_UKM          "ukm"       /* octet string */
+#define OSSL_KDF_PARAM_CEK_ALG      "cek_alg"   /* utf8 string */
+#define OSSL_KDF_PARAM_SCRYPT_N     "n"         /* uint64_t */
+#define OSSL_KDF_PARAM_SCRYPT_R     "r"         /* uint32_t */
+#define OSSL_KDF_PARAM_SCRYPT_P     "p"         /* uint32_t */
+#define OSSL_KDF_PARAM_MAXMEM       "max_memory" /* uint64_t */
+#define OSSL_KDF_PARAM_INFO         "info"      /* octet string */
+#define OSSL_KDF_PARAM_SEED         "seed"      /* octet string */
+#define OSSL_KDF_PARAM_SSHKDF_XCGHASH  "ssh_xcghash" /* octet string */
+#define OSSL_KDF_PARAM_SSHKDF_SESSION_ID "ssh_session_id" /* octet string */
+#define OSSL_KDF_PARAM_SSHKDF_TYPE  "ssh_type"  /* int */
```

Comment:
So for SSHKDF, the strings have prefixes ("ssh_"), but not for SCRYPT...  Seeing that the previous implementation didn't use prefixes in the strings, you might want to reconsider.  Consistency might not be a bad thing (and if you want to stay compatible with pre-3.0 OpenSSL, you can't prefix the SCRYPT strings, *or* you will have to add more translates)

Reply:
Fixed, all names are back to what they were.

Except the scrypt _N_ which is now _n_ with the PKEY wrapper doing the conversion.